### PR TITLE
All tasks from HW 8

### PR DESCRIPTION
**Pull request by albina-pasichnyk**
In scope of this pull request following tasks were resolved:
Task 8.1 - Decorator with function name for summation & multiplication function
Task 8.2 - Generator of square of even numbers from the range
Task 8.3 - Defining List filtered using comprehension (divisible by 7)
Task 8.4 - Defining List filtered using comprehension (contain 3 number)

### CODE REVIEW

File: HW_8/HW_8.3_Divisible_numbers.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+def divisible_number(sequence: list) -> list:
+    """
+    Function for processing and filtering list based on specific condition: divisible by 7
+    """
+    result_list = []
+    for each in sequence:
+        if each % 7 == 0:
+            result_list.append(each)
+    return result_list
```
Comment (bohdandovbsyh) [inline HW_8/HW_8.3_Divisible_numbers.py:20]: it is a task for comprehension. Function it too much for this task

File: HW_8/HW_8.3_Divisible_numbers.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+"""
+3. Find all of the numbers from 1-1000 that are divisible by 7
+"""
+
+# Defining list variable with elements from range 1 to 1000
+numbers_list = list(range(1, 1000))
+
+# Filtering input sequence using comprehension
+processed_list = [number for number in range(1, 1000) if number % 7 == 0]
```
Comment (bohdandovbsyh) [inline HW_8/HW_8.3_Divisible_numbers.py:9]: you need only this line for a solution

File: HW_8/HW_8.4_Numbers_containing.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
+"""
+4. Find all of the numbers from 1-1000 that have a 3 in them
+"""
+
+# Defining list variable with elements from range 1 to 1000
+numbers_list = list(range(1, 1000))
+
+
+# Filtering input sequence using comprehension
+processed_list = [number for number in range(1, 1000) if '3' in str(number)]
```
Comment (bohdandovbsyh) [inline HW_8/HW_8.4_Numbers_containing.py:10]: you need only this line for a solution

**Review by bohdandovbsyh — APPROVED** [review]

File: HW_8/HW_8.3_Divisible_numbers.py (hunk 1 above)
Comment (albina-pasichnyk) [reply HW_8/HW_8.3_Divisible_numbers.py:20]: Sure. It was intended as double-check and to show that comprehension performs the same things with less amount of code.

File: HW_8/HW_8.3_Divisible_numbers.py (hunk 2 above)
Comment (albina-pasichnyk) [reply HW_8/HW_8.3_Divisible_numbers.py:9]: Sure.

File: HW_8/HW_8.4_Numbers_containing.py (hunk 3 above)
Comment (albina-pasichnyk) [reply HW_8/HW_8.4_Numbers_containing.py:10]: Yep.

**Pull request merged**